### PR TITLE
chore: Move codenav types to lower-level package

### PR DESCRIPTION
**Pull request by varungandhi-src**
For https://github.com/sourcegraph/sourcegraph/pull/64126, I need to be able to
access some of these types in the codenav package directly, so move these
to avoid an import cycle: codenav -> resolvers -> codenav.

## Test plan

Covered by existing tests

### CODE REVIEW

File: internal/codeintel/resolvers/codenav_test.go [hunk 1]
```diff
@@ -5,13 +5,15 @@ import (
 	"crypto/sha1"
 	"encoding/base64"
 	"encoding/json"
-	"golang.org/x/exp/rand"
 	"testing"
 
+	"golang.org/x/exp/rand"
+
 	"github.com/stretchr/testify/require"
```
Comment (kritzcreek) [inline internal/codeintel/resolvers/codenav_test.go:12]:
```suggestion
	"golang.org/x/exp/rand"
	"github.com/stretchr/testify/require"
```

File: internal/codeintel/codenav/types.go [hunk 2]
```diff
@@ -289,3 +300,108 @@ type RemoteCursor struct {
 	// The location offset within the associated batch of uploads.
 	LocationOffset int `json:"locationOffset"`
 }
+
+type UsagesForSymbolResolvedArgs struct {
```
Comment (kritzcreek) [inline internal/codeintel/codenav/types.go:304]: nit: Where possible I'd prefer declaring types next to their primary usage. Could this type be declared in `service.go` next to the "PreciseUsages" function that accepts it?

**Review by kritzcreek — APPROVED** [review]
LGTM, just nits

File: internal/codeintel/codenav/types.go (hunk 2 above)
Comment (varungandhi-src) [reply internal/codeintel/codenav/types.go:304]: There are a fair number of types here, and `service.go`/`service_new.go` primarily have the logic for the service methods themselves (whereas this file houses the related type definitions), so I'd rather keep the types in this file following the existing code.

Comment (varungandhi-src) [reply internal/codeintel/codenav/types.go:304]: There is a similar pattern in `codenav/shared/types.go` and `uploads/shared/types.go` where the main types involved in parameters/results and their methods are defined in a separate types.go file

**Pull request merged**